### PR TITLE
discount 2.2.7b

### PR DESCRIPTION
**Pull request by danielnachun**
`pdfpc` fails to build on Linux because `discount` is missing both the `pkg-config` files and shared libraries needed for it to be found and linked by `pdfpc`: https://github.com/Homebrew/homebrew-core/pull/118911.

This PR will fix this by enabling the `pkg-config` files and shared libraries.  The idiosyncratic build system for `discount` tries to invoke `ldconfig` on Linux which fails because doing so requires root access, so a Homebrew-specific workaround is needed to disable this behavior by calling `true` instead of `ldconfig`.

### CODE REVIEW

**Review by dawidd6 — DISMISSED** [review]

**Comment by SMillerDev** [comment]
Could you report this root requirement as an issue upstream?

File: Formula/discount.rb
Comment (fxcoudert) [inline Formula/discount.rb:47]: We should not modify the scripts. If it is an upstream bug, they can fix it. If it is the behavior upstream thinks desirable, we should keep it.

**Review by fxcoudert — CHANGES_REQUESTED** [review]
I see there is a version [2.2.7b](https://www.pell.portland.or.us/~orc/Code/discount/discount-2.2.7b.tar.bz2), could we update?

**Comment by chenrui333** [comment]
trying out discount 2.2.7b

**Comment by fxcoudert** [comment]
1. `discount: Formula version newer than livecheck`
2. I still think we should not patch the script: if upstream thinks it's necessary to call a root utility, there is probably a good reason. Or it should be raised for them to fix.

**Comment by chenrui333** [comment]
> 1. discount: Formula version newer than livecheck

yeah, I can help with that.

**Comment by danielnachun** [comment]
I've got an update for this - it turns out that `pdfpc` will build fine if we just add the `--pkg-config` flag (which doesn't require any patching) but not the `--shared` flag.  While using the shared library is preferable, it is not necessary for now, and I've updated the PR so that it adds the missing `pkg-config` files so that dependents can find the static library.

> I still think we should not patch the script: if upstream thinks it's necessary to call a root utility, there is probably a good reason. Or it should be raised for them to fix

Before merging this I'd like to open an issue on the upstream GitHub that I will link in a comment in the formula explaining this problem.  This is a fairly unusual situation in which the upstream build script manually invokes `ldconfig` to rebuild `ld.so.cache`.  On modern Linux distros this is only done by the host package manager (i.e. `apt`, `yum`, etc.), and in Homebrew, we rely on RPATH to find libraries rather than `ld.so.cache`.   Manually invoking it like they do in their build script could actually cause breakage and I suspect they did not understand that there are better ways to find the shared library than messing with `ldconfig`.

**Comment by danielnachun** [comment]
I opened the issue upstream and added the comment - this should be ready to merge when CI is done.

**Review by SMillerDev — APPROVED** [review]

**Review by iMichka — APPROVED** [review]

**Review by fxcoudert — APPROVED** [review]

**Pull request closed without merging**
